### PR TITLE
Ensure footnotes with equations aren't dropped

### PR DESCRIPTION
**Pull request by rnathuji**
This issue was discovered in Organizational Behavior where a footnote
link was broken due to the fact that the content was dropped during
transformation from CNXML to HTML. There is a predicate for ignoring
`<para>` elements when they contain `<equation>`s so the corresponding
HTML does not contain blockish elements as children. However, when
contained in a footnote, these elements get pulled out and wrapped
within an `<aside>` anyways so the existing check is overly strict.
This change adds an exception for these cases so equations can be
contained within `<para>` elements so long as they are contained
within a footnote.

### CODE REVIEW

File: rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl [hunk 1]
```diff
@@ -275,7 +275,7 @@
 
 <!-- Help ensure that HTML paragraphs do not contain blockish elements as children -->
 
-<xsl:template match="c:para[not(.//c:figure|.//c:list[not(@display='inline')]|.//c:table|.//c:media[not(@display) or @display='block']|.//c:equation|.//c:preformat|.//c:note|.//c:exercise)]" name="convert-para">
+<xsl:template match="c:para[not(.//c:figure|.//c:list[not(@display='inline')]|.//c:table|.//c:media[not(@display) or @display='block']|.//c:equation[not(ancestor::c:footnote)]|.//c:preformat|.//c:note|.//c:exercise)]" name="convert-para">
```
Comment (tomjw64) [inline rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl:278]: I'm not sure this is resilient to a structure like: `footnote -> para -> equation` even though it's not intended for that case. Maybe theres a crafty xpath selector that ensures the footnote is a child of the para as well to enforce a `para -> footnote -> equation`-like structure?

**Comment by therealmarv** [comment]
@rnathuji does `./script/test` run successfully for you?
(I use a Ubuntu 16.04 bento Vagrant box for this because it tires to run homebrew on Mac)

I get errors on `master` and `cnx1102/fix-footnote-equations` and even after running `./script/rebuild-tests` which is not good.

Maybe a CI test would help here too for the future.

**Comment by rnathuji** [comment]
@therealmarv - Hmmm...they run fine for me in a 18.04 VM. I believe I just did:

```
mkvirtualenv cnxmlutils
./script/setup
./script/test
```

For CI, are you referring to something additional beyond [this](https://github.com/openstax/rhaptos.cnxmlutils/blob/9d78f4f1dfbde353e9c37a556573b862df2fff26/.travis.yml#L13)?

File: rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl (hunk 1 above)
Comment (rnathuji) [reply rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl:278]: @tomjw64 - Is this scenario you are referring to?:

```
  <footnote>
    <para>
      <equation>x=3</equation>
    </para>
  </footnote>
```

In the above, that equation will get unwrapped [here](https://github.com/openstax/rhaptos.cnxmlutils/blob/9d78f4f1dfbde353e9c37a556573b862df2fff26/rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl#L952-L958) and you end up with the `<aside>` being pulled down to the end since it's not part of a `<para>`.

Comment (tomjw64) [reply rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl:278]: Ah I see. So it's very unlikely that a footnote could be an ancestor outside of the para of interest. That's probably good enough.

Comment (rnathuji) [reply rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl:278]: Oh, in case you mean this scenario:

```
  <para>
    Content with footnote
    <footnote id="p5">
     <para>
      Footnote with equation: <equation id="eq1" class="unnumbered">x=2</equation>
      </para>
    </footnote>
  </para>
```

That works as well due to the unwrapping (produces the same output within the test case I added except for the comment on unwrapping). There probably is still some corner case, though, with the just the right mix of things.

**Comment by therealmarv** [comment]
I will rebuild my VM. Really strange, I got always at least 2 failing tests (don't know why)
